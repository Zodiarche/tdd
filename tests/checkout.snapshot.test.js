const InMemoryProductRepo = require("../src/services/inMemoryProductRepo");
const CartService = require("../src/services/cartService");

const seed = [{ id: "SKU1", name: "Item", price: 10, stock: 3 }];
const client = { email: "jhondoe@email.com" };

test("Snapshot de la commande", async () => {
  const repo = new InMemoryProductRepo(seed);
  const email = { sendOrderConfirmation: jest.fn().mockResolvedValue() };
  const service = new CartService(repo, email);

  service.add("SKU1", 2);
  const order = await service.checkout(client.email);
  // Normalise la date pour snapshot stable
  order.createdAt = "2025-01-01T00:00:00.000Z";
  expect(order).toMatchSnapshot();
});
