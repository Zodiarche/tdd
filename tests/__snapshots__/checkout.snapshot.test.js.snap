// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Snapshot de la commande 1`] = `
{
  "createdAt": "2025-01-01T00:00:00.000Z",
  "lines": [
    {
      "id": "SKU1",
      "name": "Item",
      "price": 10,
      "quantity": 2,
    },
  ],
  "totals": {
    "discountHT": 0,
    "ht": 20,
    "subtotalHT": 20,
    "ttc": 24,
    "vat": 4,
  },
}
`;
